### PR TITLE
DOC: Update refguide_check note on how to skip code

### PR DESCRIPTION
**Pull request by sethtroisi**
Extracted from https://github.com/numpy/numpy/pull/15419

### CODE REVIEW

File: tools/refguide_check.py [hunk 1]
```diff
@@ -981,9 +989,8 @@ def check_doctests_testfile(fname, verbose, ns=None,
     with open(fname, encoding='utf-8') as f:
         text = f.read()
 
-    PSEUDOCODE = set(['some_function', 'some_module', 'import example',
-                      'ctypes.CDLL',     # likely need compiling, skip it
-                      'integrate.nquad(func,'  # ctypes integrate tutotial
+    PSEUDOCODE = set([
+        'some_function', 'some_module', 'import example',
```
Comment (mattip) [inline tools/refguide_check.py]: Why this change?

File: tools/refguide_check.py [hunk 2]
```diff
@@ -947,6 +947,14 @@ def check_doctests_testfile(fname, verbose, ns=None,
     Notes
     -----
 
+    refguide can be signalled to skip testing code by adding '#doctest: +SKIP'
```
Comment (eric-wieser) [inline tools/refguide_check.py]:
```suggestion
    refguide can be signalled to skip testing code by adding ``#doctest: +SKIP``
```

File: tools/refguide_check.py (hunk 1 above)
Comment (sethtroisi) [reply tools/refguide_check.py]: These don't match anything in the code base. (ctypes.cdll would but it's already skipped with # doctest +skip)

File: tools/refguide_check.py (hunk 2 above)
Comment (sethtroisi) [reply tools/refguide_check.py]: Done

File: tools/refguide_check.py (hunk 1 above)
Comment (mattip) [reply tools/refguide_check.py]: I think they come from scipy, so let's leave them in

Comment (sethtroisi) [reply tools/refguide_check.py]: ^ they do, I'll revert this part of the cl.

**Comment by mattip** [comment]
Thanks @sethtroisi
